controllers/phschedule_controller.go

```diff
@@ -17,34 +17,241 @@ package controllers
 
 import (
 	"context"
+	"time"
 
 	"github.com/go-logr/logr"
+	cron "github.com/robfig/cron/v3"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	primehubv1alpha1 "primehub-controller/api/v1alpha1"
+	"primehub-controller/pkg/graphql"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type PhScheduleCron struct {
+	c          *cron.Cron
+	entryID    cron.EntryID
+	recurType  primehubv1alpha1.RecurrenceType
+	recurrence string
+	// (TODO) add timezone
+}
+
 // PhScheduleReconciler reconciles a PhSchedule object
 type PhScheduleReconciler struct {
 	client.Client
-	Log logr.Logger
+	Log               logr.Logger
+	PhScheduleCronMap map[string]*PhScheduleCron
+	GraphqlClient     *graphql.GraphqlClient
+}
+
+func (r *PhScheduleReconciler) buildPhJob(phSchedule *primehubv1alpha1.PhSchedule) (*primehubv1alpha1.PhJob, error) {
+	log := r.Log.WithValues("phschedule", phSchedule.Name)
+	t := time.Now().UTC()
+	hash, err := generateRandomString(4)
```

Comment:
In Job Submission, Canner uses 6 chars hash. Do we want to follow it?

Reply:
change to 6